engine/runtime-compiler/src/main/scala/org/enso/compiler/refactoring/IRUtils.scala

```diff
@@ -64,25 +65,39 @@ trait IRUtils {
     for {
       usages <- findDynamicUsages(ir, node)
     } yield {
-      usages
-        .collect {
-          case usage: Name.Literal
-              if usage.isMethod && usage.name == node.name =>
-            usage
-        }
-        .flatMap { symbol =>
-          symbol.getMetadata(MethodCalls).flatMap { resolution =>
-            resolution.target match {
-              case BindingsMap.ResolvedModuleMethod(module, _)
-                  if module.getName == moduleName =>
-                Some(symbol)
-              case _ =>
-                None
-            }
+      usages.collect {
+        case Application.Prefix(function: Name.Literal, args, _, _, _)
+            if function.name == node.name =>
+          function.getMetadata(MethodCalls) match {
+            case Some(resolution) =>
+              resolution.target match {
+                case BindingsMap.ResolvedModuleMethod(module, _)
+                    if module.getName == moduleName =>
+                  Some(function)
+                case _ =>
+                  None
+              }
+            case None =>
+              args.headOption match {
+                case Some(arg) if isSyntheticArgument(arg) =>
```

Comment:
Do we not set any flag that the argument is synthetic? I kind of dislike comparing the code and `<...>`

Reply:
No, the argument doesn't even have such a field.

Reply:
Now it has https://github.com/enso-org/enso/pull/11556/commits/b077070e5ec7e338244dc615d160becf5b25e667

Reply:
It's one of those "be careful what you wish for" changes